app/src/org/commcare/connect/PersonalIdManager.java

```diff
@@ -609,7 +609,27 @@ public boolean isLoggedInWithConnectApp(Context context, String appId) {
         return isloggedIn() && isConnectApp(context, appId);
     }
 
-    public static AuthInfo.TokenAuth getHqTokenIfLinked(String username) throws TokenDeniedException, TokenUnavailableException {
+    private boolean isPersonalIdLinkedApp(String appId, String username) {
+        if (isloggedIn()) {
+            ConnectLinkedAppRecord record = ConnectAppDatabaseUtil.getConnectLinkedAppRecord(
+                    manager.parentActivity, appId, username);
+            return record != null && record.getPersonalIdLinked();
+        }
+        return false;
+    }
+
+    public boolean isSeatedAppLinkedToPersonalId(String username) {
+        if (isloggedIn()) {
+            String seatedAppId = CommCareApplication.instance().getCurrentApp().getUniqueId();
+            ConnectLinkedAppRecord appRecord = ConnectAppDatabaseUtil.getConnectLinkedAppRecord(
+                    CommCareApplication.instance(), seatedAppId, username);
+            return appRecord != null && appRecord.getPersonalIdLinked();
+        }
+        return false;
+    }
```

Comment:
this should reuse `isPersonalIdLinkedApp`, please be careful of introducing duplicate code in commits.